extensions/extended-isupport.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "`extended-isupport` Extension"
+layout: spec
+copyrights:
+  -
+    name: "Simon Ser"
+    period: "2024"
+    email: "contact@emersion.fr"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `extended-isupport` capability name. Instead, implementations SHOULD
+use the `draft/extended-isupport` capability name to be interoperable with
+other software implementing a compatible work-in-progress version. The final
+version of the specification will use unprefixed capability names.
+
+## Introduction
+
+`RPL_ISUPPORT` is used to advertise metadata about the server as key-value
+entries. However, `RPL_ISUPPORT` is only sent by servers after connection
+registration. This undermines the usefulness of `RPL_ISUPPORT`: some of the
+metadata would be useful to clients prior to connection registration. This
+extension adds a mechanism to send `RPL_ISUPPORT` messages before connection
+registration, as well as an end of `RPL_ISUPPORT` list indication.
+
+## Implementation
+
+### `draft/extended-isupport` capability
+
+When the `draft/extended-isupport` capability is enabled by the client, the
+server MUST send one or more `RPL_ISUPPORT` messages (grouped inside a
+`draft/isupport` batch if the `batch` capability is enabled). While
```

Comment:
It might be an idea to spec this up in its own section so it can be used for other 005 responses.

Reply:
The intent is that the section about the batch covers this, specifically:

> The server MUST group all `RPL_ISUPPORT` messages inside a `draft/isupport`
batch when the `batch` and `draft/extended-isupport` capabilities are
enabled.

Reply:
Fair enough I must have missed that.

Reply:
It's mentioned in the introduction that this is sent before registration, but I think this section is where it needs to be explicit. Also maybe worth using wording to clarify that this is sent before capability negotiation is completed with CAP END.